proposed/standard-ci-env-variables.md

```diff
@@ -0,0 +1,48 @@
+
+The .NET Core SDK is implementing new scenarios that require source control manager information. An initial idea was to call out to specific source control tools to get this information. This approach is problematic since it requires an implementation for each source control manager and it has the potential to be fragile and slow.
+
+It turns out that CI services provide all of the information that these new .NET Core scenarios need via environment variables. Also, there is a strong relationship between these scenarios and official CI-provided builds (much less on local developer builds). As a result, relying on the CI-provided environment variables is attractive.
+
+To make it possible to provide application platform provided experiences across CI services, we need a standardized set of environment variables that are supported across those same CI services. Ideally, this set of environment variables would be supported across multiple CI services and useful for multiple application environments, not just .NET.
+
+## Proposed environment variables
+
+The .NET Core SDK needs are oriented around source control. As a result, the intial list is source control oriented, but there is no affinity to source control on the general idea of standardized environment variables.
+
+It is important that these environment variables do not conflict with other variables. To avoid that, all environment variables will be prepended with "STANDARDCI\_". This name is a first proposal for the idea and it may get changed based on feedback.
+
+* **STANDARDCI\_REPOSITORYCOMMITID** -- Commit hash / ID; Example: 2ba93796dcf132de447886d4d634414ee8cb069d
+* **STANDARDCI\_REPOSITORYROOT** -- Root of repository; Example: D:\repos\corefx
+* **STANDARDCI\_REPOSITORYNAME** -- Name of repository; Example: dotnet\corefx
+* **STANDARDCI\_REPOSITORYURI** -- Uri for repository; Example: https://github.com/dotnet/corefx
+
+## Support from CI Services
+
+This plan will only work if CI services decide to support these environment variables. An important question is whether CI services have similar environment variables already. The table below suggests that the information we need is already available. An arbitrary sample of CI services were picked for this exercise.
+
+| Environment Variable | VSTS | Travis CI| AppVeyor | Circle CI | AWS CodeBuild |
+| -------------------- | ---- | -------- | -------- | --------- | ------------- |
+|STANDARDCI\_REPOSITORYCOMMITID | BUILD\_SOURCEVERSION | TRAVIS\_COMMIT |APPVEYOR\_REPO\_COMMIT | CIRCLE\_SHA1 | CODEBUILD_RESOLVED_SOURCE_VERSION |
+|STANDARDCI\_REPOSITORYROOT|BUILD\_REPOSITORY\_LOCALPATH|TRAVIS\_BUILD\_DIR| APPVEYOR\_BUILD\_FOLDER | CIRCLE\_WORKING\_DIRECTORY | CODEBUILD_SRC_DIR |
+|STANDARDCI\_REPOSITORYNAME|BUILD\_REPOSITORY\_NAME| TRAVIS\_REPO\_SLUG | APPVEYOR\_REPO\_NAME | CIRCLE\_PROJECT\_USERNAME + CIRCLE\_PROJECT\_REPONAME |
+|STANDARDCI\_REPOSITORYURI|BUILD\_REPOSITORY\_URI| | | CIRCLE\_REPOSITORY\_URL | CODEBUILD_SOURCE_REPO_URL |
```

Comment:
In case of [TeamCity](https://www.jetbrains.com/teamcity/) you have the [following build parameters](https://confluence.jetbrains.com/display/TCDL/Predefined+Build+Parameters) which could be used during the build:

`STANDARDCI_REPOSITORYCOMMITID` -> `build.vcs.number`
`STANDARDCI_REPOSITORYROOT` -> `teamcity.build.checkoutDir`
`STANDARDCI_REPOSITORYNAME` -> `vcsroot.username` + `env.TEAMCITY_PROJECT_NAME`
`STANDARDCI_REPOSITORYURI` -> `vcsroot.url`

Reply:
Yet again this only works in case of a single source control root.

Reply:
Thanks @dtretyakov. Added to doc.

Reply:
@dtretyakov: what is the idea around `vcsroot.username`  +  `env.TEAMCITY_PROJECT_NAME`? The first is an empty string in our builds, and the second is just the name of the parent entity in TeamCity hierarchy.